Ginger/GingerCoreNETUnitTest/GingerRunnerTests/GingerRunnerTest.cs

```diff
@@ -265,7 +285,83 @@ public void TestRunsetConfigBFVariables()
 
         }
 
-        
+
+        [TestMethod]
+        public void TestDyanamicRunsetOprations()
+        {
+            RunSetConfig runSetConfigurations = CreteRunsetWithOperations();
+
+            GMR = new RunsetExecutor();
+            GMR.RunsetExecutionEnvironment = environment;
+            GMR.RunSetConfig = runSetConfigurations;
+
+            CLIHelper cLIHelper = new CLIHelper();
+            cLIHelper.RunAnalyzer = true;
+            cLIHelper.ShowAutoRunWindow = false;
+            cLIHelper.DownloadUpgradeSolutionFromSourceControl = false;
+
+            RunSetAutoRunConfiguration autoRunConfiguration = new RunSetAutoRunConfiguration(solution, GMR, cLIHelper);
+            CLIDynamicXML mCLIDynamicXML = new CLIDynamicXML();
+            autoRunConfiguration.SelectedCLI = mCLIDynamicXML;
+            string file = autoRunConfiguration.SelectedCLI.CreateContent(solution, GMR, cLIHelper);
+
+            XElement nodes = XElement.Parse(file);
+
+            List<XElement> AddRunsetOPerationsNodes = nodes.Elements("AddRunsetOperation").ToList();
+
+            XmlDocument xDoc = new XmlDocument();
+            xDoc.LoadXml(file);
+            XmlNodeList runsetoperations = xDoc.GetElementsByTagName("AddRunsetOperation");
+
+            //Assert
+            Assert.AreEqual(runsetoperations.Count , 3);
+            Assert.AreEqual(runsetoperations[0].FirstChild.Name, "MailFrom");
+            Assert.AreEqual(runsetoperations[0].LastChild.Name, "IncludeAttachmentReport") ;
+            Assert.AreEqual(runsetoperations[1].FirstChild.Name, "selectedHTMLReportTemplateID");
+            Assert.AreEqual(runsetoperations[1].LastChild.Name, "isHTMLReportPermanentFolderNameUsed");
+            Assert.AreEqual(runsetoperations[2].HasChildNodes,false);
+
+        }
+
+
+        public RunSetConfig CreteRunsetWithOperations()
+        {
+            runSetConfig = new RunSetConfig();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make the enclosing instance method 'static' or remove this set on the 'static' field.](https://app.codacy.com/app/Ginger-Automation/Ginger/pullRequest?prid=4629760)